src/main/java/gregtech/api/util/StreamUtils.java

```diff
@@ -0,0 +1,103 @@
+package gregtech.api.util;
+
+import net.minecraft.item.*;
+import net.minecraftforge.items.*;
+
+import java.util.*;
+import java.util.function.*;
+import java.util.stream.*;
+
+/**
+ * Various Quality-of-Life methods for working with Java 8 Streams.
+ */
+public class StreamUtils
```

Comment:
I this should be separated to two files. Because mixing highly abstract logic and concrete one in file is hard on readers. Please move ItemsStack specific logic to own file.

Reply:
Addressed in 1208289